### ISSUE TITLE
Kernel 4.9 for nile platform

### ISSUE THREAD

**Issue by thiscantbeserious** [PixelExperience-Devices/device_sony_customization#17]
Hi,

Kernel 4.14 seems to have broken camera since some months on pioneer at least. Also there are various issues with slow LTE speed and low microphone sound. I was about to give up but I happened to stumble upon this almost clean ASOP rom using Kernel 4.9 to workaround these issues, seemingly 4.9 is STABLE on nile much unlike 4.14.

ASOP ROM
https://forum.xda-developers.com/xa2-ultra/development/rom-aosp-t4002791

Official Bug Report / Issue on Sony bugtracker:
https://github.com/sonyxperiadev/bug_tracker/issues/574

Same is true for any camera on current build like Open Camera - not even using camera2 API (can't connect to camera).

Could you evaluate of switching back to Kernel 4.9 for Nile until things are stable for Pixel Experience? So far it's far from it.

Sorry if that's the wrong place for this issue.

Regards

**Comment by thiscantbeserious** [comment]
ping @kholk

**Comment by thiscantbeserious** [comment]
@MarijnS95 in reaction to your thumbs down and confused smiley:

The ping above was since I was under the impression that I've opened up this issue in kholik's personal repository in regards to PixelExperience but noticed that it opened it up in the general one - so I wasn't sure who to address this too. 

The Telegram Channel is full of people crying for help, not even reading the response and the Bot is trying to keep control.

I tried multiple times to get any information where I should submit this info but never got an answer how I should proceed with my findings. 

I'm a bit confused how to handle device specific issue like here in regards to Pixel Experience 

and honestly there's no guidline on the PixelExperience webpage either. Github is far less noisy, and most projects I know have no troubles being mentioned but rather proceed with a constructive discussion - hence I went to GH and looked for the most fitting repository related.

Sorry if I offended / confused you.

**Comment by shank03** [comment]
Are you in Pixel experience group for Sony ?

**Comment by thiscantbeserious** [comment]
Thanks - that's helpful - I didn't know such a thing existed.

**Comment by MarijnS95** [comment]
> and most projects I know have no troubles being mentioned but rather proceed with a constructive discussion

I don't consider a kernel downgrade - especially a single device with "minor" issues - constructive at all. In case you aren't aware, this means going through hell and beyond to switch to older HALs and a different device configuration. That's really hard to set up as all Sony devices are built from the same tree.
(I suspect discussing what solutions may or may not be viable constitutes to "proceed with a constructive discussion"?)

In fact @kholk - who is also the main kernel developer for Sony Open Devices - is right on top of LTE speed issues on Ninges, fixing it just yesterday https://github.com/sonyxperiadev/kernel/pull/2324.

Same for the camera; other devices are not affected meaning a likely local configuration mistake. Finding and fixing this should take a lot less time and repeated breakage than downgrading _everything_.

Finally, the microphone: when did this start happening? We recently upgraded audio and perhaps Nile is missing part of the configuration; I have no idea whether my commits made it in the last PE build. Discovery seems fine, at least on an audio recorder. Perhaps you need to clarify _where_ the microphone volume is too low: calls? Video recording? Audio recording? Wired headphones?
I remember a similar issue occurring long ago (on 4.9), iirc also on Nile - plugging in USB solved it because a regulator went down otherwise. With more detailed debug info we might as well solve this issue rather than wasting time on a massive downgrade.

TL;DR: Create **DETAILED** issues and bump existing reports with relevant information instead of requesting maintainers to undo all their work. That benefits our response times, too :wink:

**Comment by MarijnS95** [comment]
For completeness, the sdm660 mic bias issue I was referring to:
https://github.com/sonyxperiadev/kernel/pull/2174
https://github.com/sonyxperiadev/kernel-techpack-audio/pull/15

**Comment by thiscantbeserious** [comment]
Thank you for taking the time to write that up - you are right in that it creates additional noise and it certainly wasn't my intention, so sorry for that.

I'm not a Kernel Developer myself - as are likely most of your users. That's likely why I didn't see a lot of things myself that you likely think of as natural from your side of things.

I think its more or less a issue of having no guideline what to do - for people that visit the webpage and seek for the infos there's next to nothing for many devices (besides the Telegram channel / bot which is somewhat to late in my opinion and will often create additional noise)

- FAQ
- Guideline of Device-Specific Issues
- Webform that enables people to submit logs which gets pushed to a GitHub repo
- And so on ...

Is the webpage open-source'd - as in can we (I) contribute to the documentation?

**Comment by MarijnS95** [comment]
> I'm not a Kernel Developer myself - as are likely most of your users. That's likely why I didn't see a lot of things myself that you likely think of as natural from your side of things.

Hence the explanation :wink: 

> I think its more or less a issue of having no guideline what to do

I have no idea what guide PE provides for building, debugging and contributing. For what it's worth I'm only involved in the Sony Open Devices Project, with a few of my SODP development mates maintaining the PE port based on that. PE is a significant userbase making it valuable to pull/correlate issues and reports from, as well as push stable-ish experiments @kholk and me are conducting every once in a while.

In my personal case, and I'm sure it applies to the others this (SODP) is a project I contribute to (maintain, pretty much) in my free time. While documentation efforts are ongoing by some members, if I were tasked with that there'd be no time for features nor fixes. Somewhat harsh, but I'd rather leave the development side of this project for a select few interested souls with lots of prior experience (or lots of time and enthusiasm to dive right in, which is how we all ended up here) than spend my limited time documenting and updating an otherwise relatively fast moving project.
@ix5 is currently on that mundane task, see and appreciate all the progress made here: https://sx.ix5.org/ and https://sx.ix5.org/opendevices.

The only thing I can tell you as sort of a FAQ, or rather "steps to report an issue" is this:
- If it seems hardware related (ie. other devices on PE not affected): camera broken, audio, nfc, connectivity, etc., report it over at SODP https://github.com/sonyxperiadev/bug_tracker/;
- Always include logs: not all developers own all devices or have it close, nor may be able to easily repro the issue in specific circumstances. PE must have some docs on that, otherwise: `adb logcat -b kernel,default`, pipe that to a file. `adb shell dmesg` in case you don't feel like annoying @kholk :smiling_imp:.
